pkg/kubelet/volumes.go

```diff
@@ -250,3 +303,22 @@ func (kl *Kubelet) newVolumeCleanerFromPlugins(kind string, name string, podUID
 	glog.V(3).Infof("Used volume plugin %q for %s/%s", plugin.Name(), podUID, kind)
 	return cleaner, nil
 }
+
+func (kl *Kubelet) newVolumeDetacherFromPlugins(kind string, name string, podUID types.UID) (volume.Detacher, error) {
+	plugName := util.UnescapeQualifiedNameForDisk(kind)
+	plugin, err := kl.volumePluginMgr.FindAttachablePluginByName(plugName)
+	if err != nil {
+		return nil, fmt.Errorf("can't use volume plugins for %s/%s: %v", podUID, kind, err)
+	}
+	if plugin == nil {
+		// Not found but not an error.
+		return nil, nil
+	}
+
+	detacher, err := plugin.NewDetacher(name, podUID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to instantiate volume plugin for %s/%s: %v", podUID, kind, err)
+	}
+	glog.V(3).Infof("Used volume plugin %q for %s/%s", plugin.Name(), podUID, kind)
```

Comment:
This message is pretty generic. Maybe worth differentiating between unmounter and detacher so it's easier to debug.